### ISSUE TITLE
p:data with text/plain

### ISSUE THREAD

**Issue by alexmilowski** [ndw/xmlcalabash1#154]
Starting with version 1.0.17, the following:

<p:data wrapper="c:query" href="query.xq" content-type="text/plain"/>

returns base64 encoded content instead of text:

<c:query xmlns:c="http://www.w3.org/ns/xproc-step" c:content-type="application/octet-stream" c:encoding="base64">ZGVjbGFyZSBuYW1lc3BhY2Ugcz0iaHR0cDovL3dlYXRoZXIubWlsb3dza2kuY29tL1YvQVBSUy8i...

**Comment by alexmilowski** [comment]
The pipeline: 

```
<p:declare-step xmlns:p="http://www.w3.org/ns/xproc"
    xmlns:c="http://www.w3.org/ns/xproc-step" version="1.0"
   >

<p:output port="result" primary="true"/>

<p:identity>
<p:input port="source">
<p:data wrapper="c:data" content-type="text/plain" href="foo.xq"/>
</p:input>
</p:identity>

</p:declare-step>
```

**Comment by alexmilowski** [comment]
The referenced file 'foo.xq':

```
element p { "hello world! }
```

**Comment by alexmilowski** [comment]
Expected output:

```
<c:data xmlns:c="http://www.w3.org/ns/xproc-step" content-type="text/plain">element p { "hello world! }
</c:data>
```

**Comment by alexmilowski** [comment]
The result in both 1.0.17-95 and 1.0.18-95:

```
<c:data xmlns:c="http://www.w3.org/ns/xproc-step" content-type="application/octet-stream" encoding="base64">ZWxlbWVudCBwIHsgImhlbGxvIHdvcmxkISB9Cg==
</c:data>
```

**Comment by phtyson** [comment]
I found that if the href value ends in '.txt' it will produce text/plain content; otherwise base64. Using 1.0.18-95.

**Comment by alexmilowski** [comment]
Huh.  Extensions should not be used to guess the media type when I explicitly give the media type of the referenced URI.

**Comment by ndw** [comment]
I think this is fixed.

**Comment by ndw** [comment]
I believe this is fixed in XML Calabash 1.0.19, released today.
